File: GMUtilsFlutter/gmutils_flutter/lib/zgmutils/data_utils/utils/result.dart
import '../../utils/collections/string_set.dart';

class Result<T> {
  final T? result;
  final StringSet? message;
  final Map<String, dynamic>? extra;

  Result(this.result, {this.message, this.extra});

  @override
  String toString() {
    return 'Result{result: $result, message: $message, extra: $extra}';
  }
}

File: GMUtilsAndroid/GMUtils/GMUtils/src/main/java/gmutils/Result.java
package gmutils;

public class Result<T> {
    public final T value;
    public final StringSet error;

    public Result(T value) {
        this(value, (StringSet) null);
    }

    public Result(T value, CharSequence error) {
        this(value, error == null ? null : new StringSet(error));
    }

    public Result(T value, StringSet error) {
        this.value = value;
        this.error = error;
    }

    @Override
    public String toString() {
        return "Result{" +
                "value=" + value +
                ", error=" + error +
                '}';
    }
}
